runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -430,7 +440,7 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 
 	bool isLambda = false;
 	if (context->isClassAnon() || context->isClassHidden()) {
-		BuildResult res = handleAnonClassName(classFileParser->getParsedClassFile(), &isLambda);
+		BuildResult res = handleAnonClassName(classFileParser->getParsedClassFile(), &isLambda, context->hostPackageName(), context->hostPackageLength());
```

Comment:
I see for hidden classes, package name is already there. So I think package name should only be added for classes defined by Unsafe.defineAnonymous().  Hidden class can be anonymous, so package name should be added for non-hidden anon classes, i.e. isClassAnon() && !isClassHidden. You can change handleAnonClassName() to take an addition parameter like `bool isHidden`, and check this parameter to decide whether to add package name.

Reply:
Probably it is working testing if openjdk adds packagename to hidden classes using your TestAnonName.class

 ```
byte[] bytes = Files.readAllBytes("TestAnonName.class");
 Class<?> hc = lookup().defineHiddenClass(bytes, false).lookupClass();
System.out.println(hc.getName());
```

Reply:
Output of Hotspot JDK15 is:
```
Hidden class name: HiddenNamedClass/0x0000000800b8f440
```

Output with my change is the same. Hidden classes don't seem to come with a host class when being created, so it shouldn't need an extra variable.

Reply:
OK. Thanks.